spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/pre/PreDecorationFilter.java

```diff
@@ -149,6 +152,9 @@ else if (!xforwardedfor.contains(remoteAddr)) { // Prevent duplicates
 					}
 					ctx.addZuulRequestHeader("X-Forwarded-For", xforwardedfor);
 				}
+				if (this.properties.isAddHostHeader()) {
```

Comment:
So maybe instead of a boolean this could be the header name (defaults to "X-Forwarded-Host")? If empty do nothing.